bundler/lib/bundler/compact_index_client/updater.rb

```diff
@@ -80,38 +84,30 @@ def update(local_path, remote_path, retrying = nil)
         FileUtils.remove_file(local_temp_path) if File.exist?(local_temp_path)
       end
 
-      def etag_for(path)
-        sum = checksum_for_file(path)
-        sum ? %("#{sum}") : nil
+      def etag_for(digests)
+        %("#{digests[:MD5].hexdigest}")
+      end
+
+      def digest_io(io, digests)
+        Gem::Package::DigestIO.new(io, digests)
       end
 
       def slice_body(body, range)
         body.byteslice(range)
       end
 
       def checksum_for_file(path)
```

Comment:
can this method be removed now that `etag_for` uses the output from `digest_io` ?